Firestore/Source/Model/FSTFieldValue.h

```diff
@@ -40,6 +41,30 @@ typedef NS_ENUM(NSInteger, FSTTypeOrder) {
   FSTTypeOrderObject,
 };
 
+/** Defines the return value for pending server timestamps. */
+typedef NS_ENUM(NSInteger, FSTServerTimestampBehavior) {
+  FSTServerTimestampBehaviorDefault,
+  FSTServerTimestampBehaviorEstimate,
+  FSTServerTimestampBehaviorPrevious,
+};
+
+/** Holds properties that define field value deserialization options. */
+@interface FSTFieldValueOptions : NSObject
+
+@property(nonatomic, readonly) FSTServerTimestampBehavior serverTimestampBehavior;
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/** Creates a FSTFieldValueOptions instance that specifies deserialization behavior for pending
+ * server timestamps. */
+- (instancetype)initWithServerTimestampBehavior:(FSTServerTimestampBehavior)serverTimestampBehavior
```

Comment:
Note that many of the other classes in here have an initializer or static factory method that just takes the public type as the source value and then they perform the conversion there (e.g. geo points, timestamps).

I think you could reduce the amount of conversion code scattered throughout this CL if you centralized it here.

Reply:
I added a factory method here that takes Snapshot Options. It's a tad ugly since it special cases default options, so let me know what you think.